src/frequenz/sdk/ml/_model_manager.py

```diff
@@ -0,0 +1,137 @@
+# License: MIT
+# Copyright © 2024 Frequenz Energy-as-a-Service GmbH
+
+"""Load, update, monitor and retrieve machine learning models."""
+
+import asyncio
+import logging
+import pickle
+from asyncio import CancelledError
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Generic, TypeVar, cast
+
+from frequenz.channels.file_watcher import EventType, FileWatcher
+
+from frequenz.sdk.actor import BackgroundService
+
+_logger = logging.getLogger(__name__)
+
+T = TypeVar("T")
+
+
+@dataclass
+class _Model(Generic[T]):
+    """Represent a machine learning model."""
+
+    data: T
+    path: Path
+
+
+class ModelManager(BackgroundService, Generic[T]):
+    """Load, update, monitor and retrieve machine learning models."""
+
+    def __init__(self, model_paths: dict[str, Path]):
+        """Initialize the model manager with the specified model paths.
+
+        Args:
+            model_paths: A dictionary of model keys and their corresponding file paths.
+        """
+        super().__init__()
+        self._models: dict[str, _Model[T]] = {}
+        self.model_paths = model_paths
+        self.load_models()
+
+    def load_models(self) -> None:
+        """Load the models from the specified paths."""
+        for key, path in self.model_paths.items():
+            self._models[key] = _Model(data=self._load(path), path=path)
+
+    @staticmethod
+    def _load(path: Path) -> T:
+        """Load the model from the specified path.
+
+        Args:
+            path: The path to the model file.
+
+        Returns:
+            T: The loaded model data.
+
+        Raises:
+            FileNotFoundError: If the model file does not exist.
+        """
+        if not path.exists():
+            raise FileNotFoundError(f"The model path {path} does not exist.")
+        with path.open("rb") as file:
+            return cast(T, pickle.load(file))
```

Comment:
Oh damn, I just realized this file reading is not async and can potentially block the event loop for a non trivial amount of time, which is not ideal. Python's built-in `asyncio` doesn't support async file I/O, so we would need to use another library for this (like [aiofiles](https://pypi.org/project/aiofiles/)). I created a separate issue because this is way beyond this PR:

* https://github.com/frequenz-floss/frequenz-sdk-python/issues/929